janusgraph-codepipelines-ci/README.md

```diff
@@ -0,0 +1,57 @@
+#JanusGraph CodePipelines CI
+CodePipelines CI is a mechanism JanusGraph can use to do release testing in massively
+parallel fashion (to the extent of AWS CodePipelines and CodeBuild service limits).
+
+## Steps to kick off a build
+Follow the procedure below to run all release testing in parallel.
+This procedure requires you to have an AWS account.
+It also requires you to create two service roles in IAM: one for CodePipeline and
+one for CodeBuild.
+
+1. Get a personal access token from [GitHub](https://github.com/settings/tokens) with repo and admin:repo_hook scopes.
+2. Navigate to the [AWS Console](https://console.aws.amazon.com) and create an IAM User with the following managed policies: AmazonS3FullAccess, AWSCodePipelineFullAccess, AWSCodeBuildAdminAccess.
+3. For this user, create security credentials and then register them in the `code-pipelines` profile on your computer with `aws configure --profile code-pipelines`.
+4. Create a service role for CodeBuild as described [here](http://docs.aws.amazon.com/codebuild/latest/userguide/setting-up.html#setting-up-service-role).
+5. Create an IAM policy for a CodePipelines service role as described [here](http://docs.aws.amazon.com/codepipeline/latest/userguide/iam-identity-based-access-control.html#view-default-service-role-policy).
+6. Create a service role for CodePipelines by using the policy you created in step 5 and then set the STS trust to codepipeline.amazonaws.com, like what you did for CodeBuild.
```

Comment:
While the AWS CLI can be used to automate these steps, it is preferable that the user manage all IAM configuration.